build.xml

```diff
@@ -988,4 +988,11 @@
 
   <target name="pre-checkin" depends="clean, default, compile-tests, lint"/>
   <target name="all" depends="jar, test, javadoc, client" />
+
+  <target name="build-successful"
+          depends="compile"
+          description="Mark the build as successful">
+    <touch file="${build.dir}/successful-build" />
```

Comment:
We don't need to remove it to match current behaviour.  After running buck in any project, the build-successful file is there, and when buck is rebuilt with ant, it stays there even if the build fails.  It is only removed when buck is updated to a new revision from git.

However I think it would make sense to remove it unconditionally before trying to build buck with ant anyway, so when the build fails a more sensible error message is shown when running buck.